service/service.go

```diff
@@ -116,20 +116,27 @@ func (s *Service) loop(ctx context.Context, roundToResume *round) error {
 		unlock := lockOSThread(ctx, roundTidFile)
 		err := round.RecoverExecution(ctx, end, s.cfg.TreeFileBufferSize)
 		unlock()
+		cleanupRound := func(err error) {
+			eg.Go(func() error {
+				if err := round.Teardown(ctx, err == nil); err != nil {
+					logger.Warn("round teardown failed", zap.Error(err), zap.Uint("epoch", round.epoch))
+				}
+				return nil
+			})
```

Comment:
Would it make sense to make this a method? Especially since further below the same code is called again (without the `cleanupRound` shell?

Reply:
I'm not sure, the method would need to return:
```go
func() error {
				if err := round.Teardown(ctx, err == nil); err != nil {
					logger.Warn("round teardown failed", zap.Error(err), zap.Uint("epoch", round.epoch))
				}
				return nil
			}
}
```
to be scheduled on the local `eg`.